src/dams_mcda/server.R

```diff
@@ -623,91 +625,89 @@ server <- function(input, output, session) {
-
-			#  for (id in criteria_inputs){
-			#    input_name <- paste(id, toString(row_id), sep='')
-			#    value <- input[[input_name]]
-			#    r[[id]] <- value
-
-			#    if (is.null(value)){
-			#      # debug nulls, doesn't modify data
-			#      message('input ', input_name, " isNull ")
-			#    }
-			#  }
-
-			#  alternatives[[row_id]] <- unlist(r) # we want in c and not list
-			#}
-			#alternatives <- unlist(alternatives)
-
-			# -------------------------------NEED TO REWRITE BY DAM------------------------------#
+			#for alternatives in tables/graphs, this generates a blank vector with labels
+			alternatives <- vector("list", length(available_alternatives))
+			for (row_id in 1:length(available_alternatives)){
+			  # for each criterion in alternatives
+			  r <- vector("list", length(available_alternatives))
+
+			  for (id in criteria_inputs){
+			    input_name <- paste(id, toString(row_id), sep='')
+			    value <- input[[input_name]]
+			    r[[id]] <- value
+
+			    if (is.null(value)){
+			      # debug nulls, doesn't modify data
+			      message('input ', input_name, " isNull ")
+			    }
+			    alternatives[[row_id]] <- unlist(r)
+			  }
+			  alternatives <- unlist(alternatives)
+
+			# -------------------------------------------------------------#
 			# assign values in new matrix
-			#raw_scores <- getRawScores()
-			#RawCriteriaMatrix <- data.frame(
-			#	matrix(raw_scores, nrow=length(available_dams), byrow=length(criteria_inputs))
-			#)
-			#message("RawCriteriaMatrix", RawCriteriaMatrix)
+			raw_scores <- getRawScores()
+			RawCriteriaMatrix <- data.frame(
+				matrix(raw_scores, nrow=length(available_dams), byrow=length(criteria_inputs))
+			)
+			message("RawCriteriaMatrix", RawCriteriaMatrix)
 
 			## assign table row, column names
-			#row.names(RawCriteriaMatrix) <- dam_names
-			#colnames(RawCriteriaMatrix) <- criteria_names
+			row.names(RawCriteriaMatrix) <- dam_names
+			colnames(RawCriteriaMatrix) <- criteria_names
 
 			## origial scores in table form
 			## for debugging table size
-			#output$FilledCriteriaTable <- renderTable(RawCriteriaMatrix, rownames=enable_rownames)
+			output$FilledCriteriaTable <- renderTable(RawCriteriaMatrix, rownames=enable_rownames)
 
 			##----------------------------------------
 			## Call WSM and format response
 			##----------------------------------------
 			## matrix setup
-			#matrix_cols <- length(criteria_inputs) # 14 default (output size, adds summedscore)
-			#matrix_rows <- length(available_dams) # 8 default
-			#matrix_levs <- length(available_alternatives) #5 default
-
-			#WeightedScoreMatrix <- array(data=NA, c(8,14,5))
-			#WeightedScoreMatrix <- round(WeightedScoreMatrix,3)
+			matrix_cols <- length(criteria_inputs) # 14 default (output size, adds summedscore)
+			matrix_rows <- length(available_dams) # 8 default
+			matrix_levs_ind <- length(available_alternatives) # 5 default
+			matrix_levs <- length(1:995) #multi-dam alternatives

```

Comment:
why 995

Reply:
levels = scenarios or multi-dam permutations given 5 decision alternatives and 8 dams.